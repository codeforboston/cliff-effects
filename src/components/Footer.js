import React from 'react';

import {
  Grid,
  Header,
  Icon,
  Segment,
} from 'semantic-ui-react';

const Footer = () => {
  return (
    <Segment
      inverted
      vertical
      style={{ padding: '2em 0em' }}
      color='teal'>
      <Grid
        container
        divided
        inverted
        stackable>
        <Grid.Row>
          <Grid.Column width={7}>
            <Header
              as='h4'
              inverted>Cliff Effects Tool
            </Header>
            <p>Made with <Icon
              name='heart'
              size='small' /> by Code for Boston
            </p>
          </Grid.Column>
        </Grid.Row>
      </Grid>
    </Segment>
  );
};

export default Footer;