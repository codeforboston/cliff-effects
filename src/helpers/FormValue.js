/** Handles the core of form values, including:
 * - Storing both a current and future value
 * - Tracking when the user manipulates a future value
 * - Propagating current values to future values if they
 * haven't been manipulated
 * - Making this instance serializable
 * 
 * All values stored in a FormValue instance should be valid
 * 
 * There must be a better way to write that.
 * 
 * @class
 * @param {string} name - the name of the client property
 * @param {any} val - the initial value
 * @param {function} [makeValid] - transforms a given value into
 * a valid value
 * 
 * @returns FormValue
 * @todo Finish description of FormValue return value
 */
class FormValue {

  constructor ( name, val, makeValid ) {

    var formVal = this;

    formVal.name = name;
    // Is there a function to set this, or just `formVal.futureChanged = true;`?
    formVal.futureChanged = false;
    // Not sure about providing this default...
    formVal.makeValid = makeValid || function ( val ) { return val; };

    formVal.setCurrent( val );

  }  // End FormValue.constructor()


  setCurrent ( val ) {

    var formVal = this;

    val = formVal.makeValid( val, 'current', formVal );

    formVal.current = val;
    if ( !formVal.futureChanged ) {
      formVal.setFuture( val )
    }

    return formVal;

  }  // End FormValue.setCurrent()


  setFuture ( val ) {
    
    var formVal = this;

    val = formVal.makeValid( val, 'future', formVal );
    formVal.future = val;
    
    return formVal;

  }  // End FormValue.setFuture()


  serialize () {

    var formVal = this;

    var obj     = {};
    obj.current = formVal.current;
    obj.future  = formVal.future;
    obj.futureChanged = formVal.futureChanged;

    return obj;
  }  // End FormValue.serialize()


  deserialize ( obj, shouldSetFuture ) {

    var formVal = new FormValue( obj.name, obj.makeValid );

    formVal.futureChanged = Boolean( obj.futureChanged );
    formVal.setCurrent( obj.current || null );

    if ( shouldSetFuture ) {
      formVal.setFuture( obj.future || null );
    }

    return formVal;

  }  // End FormValue.deserialize()

};  // End FormValue{}


export { FormValue };
