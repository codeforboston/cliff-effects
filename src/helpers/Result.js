/**
* All benefit programs' return value should be an instance of Result class.
*
* @todo Implement `.expirationDate`to track  when data needs to be update
* @external
* @class
* @param {object}
* @param {string} trial.result - 'good', 'warning', or 'bad' are the proposed
* three values, but right now they're 'good', 'information', and 'warning'
* @param {string} trial.details - Explanation of results that will be seen
* by the user. Examples: 'All good!', 'Because your income is x% above some
* value, your benefit amount could go down soon.'
* @param {string} trial.benefitValue - Monthly subsidy benefit will provide.
* @param {object} trial.data -  other information if needed. The code using your `.data` will have to know
* what to expect in there.
*/

class Result {
  constructor ( trial ) {
    this.result       = trial.result;
    this.details      = trial.details;
    this.benefitValue = trial.benefitValue;
    this.data         = trial.data;
  }
};

export { Result }
