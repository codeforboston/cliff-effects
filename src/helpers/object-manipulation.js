/** For object manipulation. Transformation? */

/** 
* Add the properties from needy to giver, overwriting existing
* properties of needy if there's a conflict.
* 
* @todo Change name so it doesn't sound like destroys both objects.
*/
const merge = function ( needy, giver ) { // Function to merge all of the properties from one object into another
    for ( var key in giver ) {  needy[key] = giver[key]; }
    return  needy;
};

/** Make the first letter of a word capitalized. */
const capitalizeWord = function ( word ) {
	return word[ 0 ].toUpperCase() + word.substr( 1 );
}


export { merge, capitalizeWord };
