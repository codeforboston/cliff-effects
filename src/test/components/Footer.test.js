import React from 'react';
import { mount } from 'enzyme';

import Footer from '../../components/Footer';
import { snippets } from '../helpers';

describe('<Footer>', () => {
  it('matches snapshot', () => {
    expect(() => {
      mount(<Footer snippets={ snippets } />);
    }).not.toThrow();
  });
});
