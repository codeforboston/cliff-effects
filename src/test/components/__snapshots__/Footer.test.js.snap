// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Footer> matches snapshot 1`] = `
<Footer
  snippets={null}
>
  <Segment
    className="footer_segment"
    color="teal"
    inverted={true}
    vertical={true}
  >
    <div
      className="ui teal inverted vertical segment footer_segment"
    >
      <Grid
        container={true}
        divided={true}
        inverted={true}
        stackable={true}
      >
        <div
          className="ui container inverted stackable divided grid"
        >
          <GridRow>
            <div
              className="row"
            >
              <GridColumn
                floated="left"
                width={4}
              >
                <div
                  className="left floated four wide column"
                >
                  <Header
                    as="h4"
                    inverted={true}
                  >
                    <h4
                      className="ui inverted header"
                    >
                      Some translated text.
                    </h4>
                  </Header>
                  <p>
                    Some translated text.
                  </p>
                </div>
              </GridColumn>
              <a
                href="http://www.codeforboston.org"
                rel="noopener noreferrer"
                target="_blank"
              >
                <Image
                  as="img"
                  floated="right"
                  size="small"
                  src="logo.svg"
                  ui={true}
                >
                  <img
                    className="ui small right floated image"
                    src="logo.svg"
                  />
                </Image>
              </a>
            </div>
          </GridRow>
        </div>
      </Grid>
    </div>
  </Segment>
</Footer>
`;
